quesma/table_resolver/rules.go

```diff
@@ -209,214 +251,168 @@ func (r *tableRegistryImpl) singleIndex(indexConfig map[string]config.IndexConfi
-			// This is because we have already checked if the pattern matches any elastic indexes.
-			for _, pattern := range input.parts {
-				for indexName := range r.elasticIndexes {
-					if util.IndexPatternMatches(pattern, indexName) {
-						return &Decision{
-							Err:    end_user_errors.ErrSearchCondition.New(fmt.Errorf("index parsedPattern [%s] resolved to elasticsearch indices", input.parts)),
-							Reason: "We're not supporting common tables for Elastic.",
-						}
-					}
+func mergeUseConnectors(lhs []ConnectorDecision, rhs []ConnectorDecision) ([]ConnectorDecision, *Decision) {
+	for _, connDecisionRhs := range rhs {
+		foundMatching := false
+		for _, connDecisionLhs := range lhs {
+			if _, ok := connDecisionRhs.(*ConnectorDecisionElastic); ok {
+				if _, ok := connDecisionLhs.(*ConnectorDecisionElastic); ok {
+					foundMatching = true
 				}
 			}
-
-			var matchedVirtualTables []string
-			var matchedTables []string
-			for _, pattern := range input.parts {
-
-				// here we check against the config
-
-				for indexName, index := range r.conf.IndexConfig {
-					if util.IndexPatternMatches(pattern, indexName) {
-
-						targets := getTargets(index, pipeline)
-
-						if slices.Contains(targets, config.ClickhouseTarget) {
-							if index.UseCommonTable {
-								matchedVirtualTables = append(matchedVirtualTables, indexName)
-							} else {
-								matchedTables = append(matchedTables, indexName)
-							}
+			if rhsClickhouse, ok := connDecisionRhs.(*ConnectorDecisionClickhouse); ok {
+				if lhsClickhouse, ok := connDecisionLhs.(*ConnectorDecisionClickhouse); ok {
+					if lhsClickhouse.ClickhouseTableName != rhsClickhouse.ClickhouseTableName {
+						return nil, &Decision{
+							Reason: "Inconsistent ClickHouse table usage",
+							Err:    fmt.Errorf("inconsistent ClickHouse table usage - %s and %s", connDecisionRhs, connDecisionLhs),
 						}
 					}
-				}
-
-				// but maybe we should also check against the actual indexes ??
-				if r.conf.AutodiscoveryEnabled {
-					for indexName, index := range r.clickhouseIndexes {
-						if util.IndexPatternMatches(pattern, indexName) {
-							if index.isVirtual {
-								matchedVirtualTables = append(matchedVirtualTables, indexName)
-							} else {
-								matchedTables = append(matchedTables, indexName)
+					if lhsClickhouse.IsCommonTable {
+						if !rhsClickhouse.IsCommonTable {
+							return nil, &Decision{
+								Reason: "Inconsistent common table usage",
+								Err:    fmt.Errorf("inconsistent common table usage - %s and %s", connDecisionRhs, connDecisionLhs),
+							}
+						}
+						lhsClickhouse.ClickhouseTables = append(lhsClickhouse.ClickhouseTables, rhsClickhouse.ClickhouseTables...)
+						lhsClickhouse.ClickhouseTables = util.Distinct(lhsClickhouse.ClickhouseTables)
+					} else {
+						if !reflect.DeepEqual(lhsClickhouse, rhsClickhouse) {
+							return nil, &Decision{
+								Reason: "Inconsistent ClickHouse table usage",
```

Comment:
The same reason is returned above. There is no way to distinguish which one rule has been fired.

Reply:
Fixed in last commit - made the reason (and corresponding Err) more clear.